File: TrainningReactJS/src/services/WorkingService.js
import axios from '../axios';

const getInforWorkingByEmployee = (id) => {
    return axios.get(`/get_working?idEmployee=${id}`);
}

const delWorkingById = (id) => {
    return axios.get(`/delete_working?idWorking=${id}`);
}

const insertWorking = (data) => {
    return axios.post(`/insert_working`, data);
}

const approvalWorking = (id) => {
    return axios.get(`/approval_working?id=${id}`);
}

export {
    getInforWorkingByEmployee,
    delWorkingById,
    insertWorking,
    approvalWorking
}

File: TrainingSV/src/main/java/com/ThanhLoc/Server/ServiceEmployees/WorkingService.java
package com.ThanhLoc.Server.ServiceEmployees;

import com.ThanhLoc.Server.domain.Working;
import com.ThanhLoc.Server.payload.Response.HourWorkingResponse;

import java.util.List;

public interface WorkingService {
    boolean insertWorking(Working working);

    List<Working> listWorking(Long id) throws RuntimeException;

    boolean deleteWorking(Long idWorking) throws RuntimeException;

    boolean deleteWorkingByEmployeeId(Long idEmployee) throws RuntimeException;

    HourWorkingResponse getHourWorking(Long id, int month, int year) throws RuntimeException;

    int approvalWorking(Long id);
}
